tests/engine/test_stats.py

```diff
@@ -35,147 +44,212 @@ def initialize_dummy_netcdf_file(name):
-        data.variables["v2"][:, -1] = 3
-
-        data.createVariable("v3", np.float64, dimensions=("t", "x"))
-        data.variables["v3"][:] = np.ones((TIME_DIM_SIZE, HOR_DIM_SIZE)) * 3
-        data.variables["v3"][:, 0] = 2
-        data.variables["v3"][:, -1] = 4
-
-        data.close()
-
-    def tear_down(self):
-        os.remove(self.nc_file_name)
-        os.remove(self.stats_file_names)
-
-    def test_stats(self):
-        file_specification = {
-            "Test data": {
-                "format": "netcdf",
-                "time_dim": "t",
-                "horizontal_dims": ["x"],
-                "fill_value_key": "_FillValue",  # should be the name for fill_value
-            },
-        }
-
-        df = create_stats_dataframe(
-            input_dir=".",
-            file_id=[["Test data", self.nc_file_glob]],
-            stats_file_name=self.stats_file_names,
-            file_specification=file_specification,
+def add_variable_to_grib(filename, dict_data):
+    with open(filename, "wb") as f_out:
+        for short_name in list(dict_data.keys()):
+            gid = eccodes.codes_grib_new_from_samples("reduced_rotated_gg_sfc_grib2")
+            eccodes.codes_set(gid, "edition", 2)
+            eccodes.codes_set(gid, "centre", "lssw")
+            eccodes.codes_set(gid, "dataDate", 20230913)
+            eccodes.codes_set(gid, "dataTime", 0)
+            eccodes.codes_set(gid, "stepRange", 0)
+            eccodes.codes_set(gid, "typeOfLevel", "surface")
+            eccodes.codes_set(gid, "level", 0)
+            eccodes.codes_set(gid, "shortName", short_name)
+            eccodes.codes_set_values(gid, dict_data[short_name])
+            eccodes.codes_write(gid, f_out)
+            eccodes.codes_release(gid)
+
+
+@pytest.fixture
+def setup_grib_file(tmp_path):
+    array_t = np.ones(
+        (
+            TIME_DIM_GRIB_SIZE,
+            STEP_DIM_SIZE,
+            HEIGHT_DIM_GRIB_SIZE,
+            HORIZONTAL_DIM_GRIB_SIZE,
         )
-
-        # check that the mean/max/min are correct
-        expected = np.array(
-            [
-                [1.0, 2.0, 0.0, 1.0, 2.0, 0.0, 1.0, 2.0, 0.0],
-                [1.0, 2.0, 0.0, 1.0, 2.0, 0.0, 1.0, 2.0, 0.0],
-                [1.0, 2.0, 0.0, 1.0, 2.0, 0.0, 1.0, 2.0, 0.0],
-                [1.0, 2.0, 0.0, 1.0, 2.0, 0.0, 1.0, 2.0, 0.0],
-                [1.0, 2.0, 0.0, 1.0, 2.0, 0.0, 1.0, 2.0, 0.0],
-                [2.0, 3.0, 1.0, 2.0, 3.0, 1.0, 2.0, 3.0, 1.0],
-                [3.0, 4.0, 2.0, 3.0, 4.0, 2.0, 3.0, 4.0, 2.0],
-            ]
+    )
+    array_t[:, :, :, 0] = 0
+    array_t[:, :, :, -1] = 2
+
+    array_pres = (
+        np.ones(
+            (
+                TIME_DIM_GRIB_SIZE,
+                STEP_DIM_SIZE,
+                HEIGHT_DIM_GRIB_SIZE,
+                HORIZONTAL_DIM_GRIB_SIZE,
+            )
         )
+        * 3
+    )
+    array_pres[:, :, :, 0] = 2
+    array_pres[:, :, :, -1] = 4
+
+    dict_data = {"t": array_pres, "v": array_t}
```

Comment:
I guess this does not change much, but why is `"t"` the key for `array_pres` and `"v"` the one for `array_t`?

Reply:
Yes, good point. I changed the variables names to be consistent.